src/main/scala/com/kinja/amqp/MessageBufferProcessor.scala

```diff
@@ -107,113 +107,35 @@ class MessageBufferProcessor(
 
 	private def processMessageBuffer(lock: Boolean)(implicit stepId: UUID, ec: ExecutionContext): Future[Unit] = {
 		logger.debug(s"[RabbitMQ][ProcessingMessageBuffer(id=$stepId)]Processing message buffer...")
-		val hasMessageF = messageStore.hasMessageToProcess()
-		val hasConfirmationF = messageStore.hasConfirmationToProcess()
-		for {
-			hasMessageToProcess <- hasMessageF
-			hasConfirmationToProcess <- hasConfirmationF
-			_ <- deleteConfirmedMessagesAndConfirmations(hasMessageToProcess, hasConfirmationToProcess)
-			_ <- deleteOrphanConfirmations(hasConfirmationToProcess)
-			_ <- if (lock) {
-				resendUnconfirmedMessages(hasMessageToProcess)
+		(for {
+			hasMessageToProcess <- messageStore.cleanup()
```

Comment:
That incorporates `deleteConfirmedMessagesAndConfirmations` and `deleteOrphanConfirmations`.